### PR TITLE
fix(Core/Spells): Resolve a case of triggered spells prolonging combat.

### PR DESCRIPTION
**Pull request by heyitsbench**
Attempt two from https://github.com/azerothcore/azerothcore-wotlk/pull/19819.

## Changes Proposed:

This PR proposes changes to:
-  [X] Core (units, players, creatures, game systems).

## Issues Addressed:

- Progresses https://github.com/azerothcore/azerothcore-wotlk/issues/13330

## SOURCE:

The changes have been validated through:
- [X] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)

## Tests Performed:

This PR has been:
- [X] Tested in-game by the author.
- [X] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [X] This pull request can be tested by following the reproduction steps provided in the linked issue

## Known Issues and TODO List:

- [ ] This PR absolutely cannot be merged until testing is done.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -20228,13 +20228,24 @@ void Unit::OutDebugInfo() const
...
 {
 public:
-    AuraMunchingQueue(Unit& owner, ObjectGuid targetGUID, int32 basePoints, uint32 spellId) : _owner(owner), _targetGUID(targetGUID), _basePoints(basePoints), _spellId(spellId) { }
+    AuraMunchingQueue(Unit& owner, ObjectGuid targetGUID, int32 basePoints, uint32 spellId, AuraEffect* aurEff, AuraType auraType) : _owner(owner), _targetGUID(targetGUID), _basePoints(basePoints), _spellId(spellId), _aurEff(aurEff), _auraType(auraType) { }
 
     bool Execute(uint64 /*eventTime*/, uint32 /*updateTime*/) override
     {
         if (_owner.IsInWorld() && _owner.FindMap())
+        {
+            bool auraFound = false;
```
Comment (sudlud) [inline src/server/game/Entities/Unit/Unit.cpp:20237]: I feel like this section could maybe need a comment describing why this search is beeing done

File: src/server/game/Entities/Unit/Unit.cpp [hunk 2]
```diff
@@ -20228,13 +20228,24 @@ void Unit::OutDebugInfo() const
...
+            Unit::AuraEffectList const& auraEffects = _owner.GetAuraEffectsByType(_auraType);
```
Comment (walkline) [inline src/server/game/Entities/Unit/Unit.cpp:20238]: I think I mixed up the owner and the target. This aura belongs to the target (right?), so we need to check the aura on the target, not the owner.

Comment (heyitsbench) [reply src/server/game/Entities/Unit/Unit.cpp]: You seem to be correct, adjusted the PR accordingly.

File: src/server/game/Entities/Unit/Unit.cpp (hunk 1 above)
Comment (heyitsbench) [reply src/server/game/Entities/Unit/Unit.cpp]: Added a comment, let me know if it's not descriptive enough.

File: src/server/game/Entities/Unit/Unit.cpp [hunk 3]
```diff
@@ -20228,13 +20228,24 @@ void Unit::OutDebugInfo() const
...
-    AuraMunchingQueue(Unit& owner, ObjectGuid targetGUID, int32 basePoints, uint32 spellId) : _owner(owner), _targetGUID(targetGUID), _basePoints(basePoints), _spellId(spellId) { }
+    AuraMunchingQueue(Unit& owner, ObjectGuid targetGUID, int32 basePoints, uint32 spellId, AuraEffect* aurEff, AuraType auraType) : _owner(owner), _targetGUID(targetGUID), _basePoints(basePoints), _spellId(spellId), _aurEff(aurEff), _auraType(auraType) { }
 
     bool Execute(uint64 /*eventTime*/, uint32 /*updateTime*/) override
     {
         if (_owner.IsInWorld() && _owner.FindMap())
             if (Unit* target = ObjectAccessor::GetUnit(_owner, _targetGUID))
-                _owner.CastCustomSpell(_spellId, SPELLVALUE_BASE_POINT0, _basePoints, target, TriggerCastFlags(TRIGGERED_FULL_MASK & ~TRIGGERED_NO_PERIODIC_RESET), nullptr, nullptr, _owner.GetGUID());
+            {
+                bool auraFound = false; // Used to ensure _aurEff exists to avoid nullptr dereference/crash
```
Comment (walkline) [inline src/server/game/Entities/Unit/Unit.cpp:20238]: This is a minor nitpick, so you can probably ignore it, but the issue we're trying to solve here is called a "dangling/wild pointer/reference".

Comment (heyitsbench) [reply src/server/game/Entities/Unit/Unit.cpp]: Don't want to be putting incorrect information in the comments. Thanks for teaching me something today! ❤️

**Comment by kissingers** [comment]
So I wonder after fix the crash, Is it still can work fine for  "resolve a case of triggered spells prolonging combat" ?

**Comment by heyitsbench** [comment]
> So I wonder after fix the crash, Is it still can work fine for "resolve a case of triggered spells prolonging combat" ?

I did test it locally and the issue I was after *was* still solved by this PR.

**Pull request merged**